x-pack/plugins/observability/public/components/shared/alert_search_bar/alert_search_bar_with_url_sync.tsx

```diff
@@ -12,13 +12,21 @@ import {
   useAlertSearchBarStateContainer,
 } from './containers';
 import { ObservabilityAlertSearchBar } from './alert_search_bar';
+import { ObservabilityAlertSearchBarProvider } from './services';
 import { AlertSearchBarWithUrlSyncProps } from './types';
+import { useKibana } from '../../../utils/kibana_react';
+import { ObservabilityAppServices } from '../../../application/types';
 
 function AlertSearchbarWithUrlSync(props: AlertSearchBarWithUrlSyncProps) {
   const { urlStorageKey, ...searchBarProps } = props;
   const stateProps = useAlertSearchBarStateContainer(urlStorageKey);
+  const services = useKibana<ObservabilityAppServices>().services;
 
-  return <ObservabilityAlertSearchBar {...stateProps} {...searchBarProps} />;
+  return (
+    <ObservabilityAlertSearchBarProvider {...services}>
```

Comment:
I would move this up to your React root.

Reply:
What is the benefit of that?

Reply:
see: https://github.com/elastic/kibana/pull/146401#discussion_r1037363463